src/conversions/std/ipaddr.rs

```diff
@@ -0,0 +1,123 @@
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+use crate::exceptions::PyValueError;
+use crate::once_cell::GILOnceCell;
+use crate::types::PyType;
+use crate::{intern, FromPyObject, IntoPy, Py, PyAny, PyObject, PyResult, Python, ToPyObject};
+
+/// A GILOnceCell that holds a reference to a Python class.
+struct PyClassRef(GILOnceCell<Py<PyType>>, &'static str, &'static str);
```

Comment:
There are a few occasions where this particular GILOnceCell pattern is used (e.g. [`types/mapping.rs`](https://github.com/PyO3/pyo3/blob/451729aef0e787a85011dc608433dea9c663ecee/src/types/mapping.rs#L118), [`conversions/rust_decimal.rs`](https://github.com/PyO3/pyo3/blob/451729aef0e787a85011dc608433dea9c663ecee/src/conversions/rust_decimal.rs#L76)). Happy to move PyClassRef outside of this module and use it consistently around the codebase if you feel that'd be an improvement. :)

Reply:
Oh, that's neat! Implemented in https://github.com/PyO3/pyo3/pull/3197/commits/721c13ce30782415606c5a06a934378eace24415. If y'all feel that's generally useful I'd be happy to apply it outside of the module. :)

Reply:
Probably easiest to centralize that in a follow-up PR, any objections to that? :)

Reply:
Please still move the method definition to `src/sync.rs` even it is only used here (the visibility is not affected by this in any case).

Reply:
Done, thanks! 😃